src/crypto/CryptState.h

```diff
@@ -7,21 +7,45 @@
 #define MUMBLE_CRYPTSTATE_H_
 
 #include "Timer.h"
+#include <chrono>
+#include <queue>
 #include <string>
 
+struct PacketStats {
+	unsigned int good   = 0;
+	unsigned int late   = 0;
+	unsigned int lost   = 0;
+	unsigned int resync = 0;
+};
+
+struct PacketStatsSnapshot {
+	PacketStats stats;
+	std::chrono::time_point< std::chrono::steady_clock > timestamp;
+};
+
 class CryptState {
 private:
 	Q_DISABLE_COPY(CryptState)
+
+	const std::chrono::seconds m_rollingScanInterval = std::chrono::seconds(5);
+	std::chrono::time_point< std::chrono::steady_clock > m_rollingTimer =
+		std::chrono::time_point< std::chrono::steady_clock >();
+
+	std::queue< PacketStatsSnapshot > m_statsLocalReference;
+	std::queue< PacketStatsSnapshot > m_statsRemoteReference;
```

Comment:
Why the use of a `std::deque`? You seem to only ever remove from these lists at the back. A `vector` can easily handle that (and is most likely more cache friendly - not that this is really important here though)

Reply:
FIFO, we add to the back and pop from the front